rebar.config.erl

```diff
@@ -268,3 +268,35 @@ str(B) when is_binary(B) -> unicode:characters_to_list(B, utf8).
 
 erl_opts_i() ->
     [{i, "apps"}] ++ [{i, Dir}  || Dir <- filelib:wildcard(filename:join(["apps", "**", "include"]))].
+
+dialyzer(Config) ->
+    {dialyzer, OldDialyzerConfig} = lists:keyfind(dialyzer, 1, Config),
+
+    AppsToAnalyse = case os:getenv("DIALYZER_ANALYSE_APP") of
+        false ->
+            [];
+        Value ->
+            [ list_to_atom(App) || App <- string:tokens(Value, ",")]
+    end,
+
+    AppsDir = "apps",
+    AppNames = [emqx | list_dir(AppsDir)],
+
+    KnownApps = [Name ||  Name <- AppsToAnalyse, lists:member(Name, AppNames)],
+    UnknownApps = AppsToAnalyse -- KnownApps,
+    io:format("UnKnown Apps ~p ~n", [UnknownApps]),
+
+    AppsToExclude = AppNames -- KnownApps,
+
+    case length(AppsToAnalyse) > 0 of
+        true ->
+            lists:keystore(dialyzer, 1, Config, {dialyzer, OldDialyzerConfig ++ [{exclude_apps, AppsToExclude}]});
+        false ->
+            Config
+    end.
+
+list_dir(Dir) ->
+    case file:list_dir(Dir) of
+        {ok, Filenames} -> [ list_to_atom(Name) || Name <- Filenames, filelib:is_dir(filename:join(["apps", Name]))];
+        {error, enoent} -> ok
```

Comment:
this should never happen as the dirs are returned from `list_dir`.
we can be assertive here:

```
{ok, Names} = file:list_dir(Dir),
[list_to_atom(Name) || Name <- Names, filelib:is_dir(filename:join([Dir, Name]))]
```